### PR TITLE
Add keystoreManager class

### PR DESCRIPTION
**Pull request by posgnu**

### CODE REVIEW

File: codechain/keystore/KeystoreManager.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        storage = list(filter(lambda storage: storage.address == key), rows)
+
+        if len(storage) == 0:
+            return None
+
+        return storage
+
+    def create_key_from_private_key(self, private_key: str, **kwargs):
+        passphrase = kwargs.get("passphrase", "")
+        meta = kwargs.get("meta", "{" + "}")
```
Comment (foriequal0) [inline codechain/keystore/KeystoreManager.py]:
```suggestion
        meta = kwargs.get("meta", "{}")
```

File: codechain/keystore/KeystoreManager.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+    def get_keys(self) -> list:
+        rows = self.db[self.key_type.value]
+        return map(lambda storage: storage.address, rows)
+
+    def import_raw(self, private_key: str, **kwargs):
+        return self.create_key_from_private_key(private_key, **kwargs)
+
+    def export_key(self, key: str, passphrase: str):
+        storage = self.get_storage
```
Comment (foriequal0) [inline codechain/keystore/KeystoreManager.py]: Shouldn't it be like this?
```suggestion
        storage = self.get_storage(key)
```

File: codechain/keystore/KeystoreManager.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    def get_storage(self, key: str):
+        rows = self.db[self.key_type.value]
+        storage = list(filter(lambda storage: storage.address == key), rows)
+
+        if len(storage) == 0:
+            return None
+
+        return storage
+
+    def create_key_from_private_key(self, private_key: str, **kwargs):
```
Comment (foriequal0) [inline codechain/keystore/KeystoreManager.py]: I think it is a private method. Should we prefix it with `_`?

File: codechain/keystore/KeystoreManager.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+from .KeyType import get_table_name
+from .KeyType import KeyType
+from .StorageJson import decode
+from .StorageJson import encode
+
+
+class keystoreManager:
+    def __init__(self, key_type: KeyType, db_path: str):
+        self.db = Database(db_path)
+        self.key_type = key_type
```
Comment (foriequal0) [inline codechain/keystore/KeystoreManager.py:19]: Why it takes `key_type`?

File: codechain/keystore/KeystoreManager.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+        data = {"meta": "", "platform": [], "asset": [], "hdwseed": []}
+        self.db.data(dictionary=data)
+
+    def get_keys(self) -> list:
+        rows = self.db[self.key_type.value]
```
Comment (foriequal0) [inline codechain/keystore/KeystoreManager.py:25]: Why it works conditionally based on `self.key_type`?

**Review by foriequal0 — CHANGES_REQUESTED** [review]

File: codechain/keystore/KeystoreManager.py (hunk 4 above)
Comment (posgnu) [reply codechain/keystore/KeystoreManager.py:19]: So we can use this class as both for asset and platform address.

File: codechain/keystore/KeystoreManager.py (hunk 5 above)
Comment (posgnu) [reply codechain/keystore/KeystoreManager.py:25]: Please check the above comment.

File: codechain/keystore/KeystoreManager.py (hunk 4 above)
Comment (foriequal0) [reply codechain/keystore/KeystoreManager.py:19]: Oh, I understood. The `jsondb` reads `db_path` every time when you update the database. I thought it reads once and writes multiple times.

File: codechain/keystore/KeystoreManager.py (hunk 1 above)
Comment (posgnu) [reply codechain/keystore/KeystoreManager.py]: I fixed it.

File: codechain/keystore/KeystoreManager.py (hunk 2 above)
Comment (posgnu) [reply codechain/keystore/KeystoreManager.py]: I fixed it.

File: codechain/keystore/KeystoreManager.py (hunk 3 above)
Comment (posgnu) [reply codechain/keystore/KeystoreManager.py]: I made it private.

**Review by foriequal0 — APPROVED** [review]

**Pull request merged**